pkg/probes/legacy/legacy.go

```diff
@@ -79,7 +79,7 @@ func (lgp Probe) GetMachineImageID(platformType string, cpuArch cpu.Architecture
 // variables listed in the template's "network-verifier-required-variables" directive, or if
 // values *are* provided for variables that must be set to a certain value for the probe to
 // function correctly (presetUserDataVariables) -- this function will fill-in those values for you.
-func (lgp Probe) GetExpandedUserData(userDataVariables map[string]string) (string, error) {
+func (lgp Probe) GetExpandedUserData(userDataVariables map[string]string, template string) (string, error) {
```

Comment:
I know it looks weird, but this is an established way in Go to signify that "Hey, I accept a parameter here to satisfy an interface, but I don't actually use it at all".

```suggestion
func (lgp Probe) GetExpandedUserData(userDataVariables map[string]string, _ string) (string, error) {
```